components/status/StatusPreviewMastodon.vue

```diff
@@ -0,0 +1,137 @@
+const timeAgoOptions = useTimeAgoOptions(true)
+const timeago = useTimeAgo(() => derivedStatus.createdAt ?? Date.now(), timeAgoOptions)
+</script>
+
+<template v-if="derivedStatus.account">
+  <div
+    flex flex-col
+    display-block of-hidden
+    bg-code
+    relative
+    w-full
+    justify-start
+    rounded-2xl
+    class="border-width-[0.05rem] border-[var(--c-text-secondary)]"
+  >
+    <div
+      p4 flex flex-col justify-between
+      basis-50 flex-auto min-h-fit max-h-fit
+    >
+      <!-- START -->
+      <!-- Account Info -->
+      <div flex basis-full flex-nowrap items-center space-x-2>
+        <div flex="~ row gap-2" grow flex-nowrap items-center justify-items-start align-baseline min-w-fit max-w-fit font-bold text-primary>
+          <AccountAvatar
+            :account="derivedStatus.account" :square="false"
+            basis-1em flex-auto max-w-5 line-clamp-1 ws-pre-wrap
+          />
+          <AccountDisplayName
+            :account="derivedStatus.account" :hide-emojis="getPreferences(userSettings, 'hideUsernameEmojis')"
+            flex-none min-w-fit max-w-fit font-bold line-clamp-1 ws-pre-wrap break-all text-primary
+          />
+          <AccountBotIndicator
+            v-if="derivedStatus.account.bot"
+            flex-none min-w-fit max-w-fit me-1
+          />
+        </div>
+        <div flex="~ row gap-2" shrink min-w-0 flex-nowrap items-start align-baseline max-w-fit class="zen-none">
+          <div flex="~ row gap-0" basis-0 grow min-w-0 flex-nowrap items-start align-baseline max-w-fit line-clamp-1 ws-pre-wrap class="zen-none">
+            <div min-w-content>
+              <span text-secondary>{{ username }}</span>
+            </div>
+            <div v-if="serverName" basis-0 grow min-w-0 max-w-fit flex-nowrap items-start align-baseline>
+              <span line-clamp-1 overflow-x-hidden text-secondary font-thin p0 m0 break-all>@{{ serverName }}</span>
+            </div>
+            <div flex-none items-start align-baseline min-w-fit line-clamp-1 class="zen-none">
+              <span text-secondary text-secondary-light> &bull; {{ timeago }}</span>
+            </div>
+          </div>
+        </div>
+        <div flex-auto />
+        <StatusActionsMore v-if="derivedStatus" :status="derivedStatus" :details="false" :command="false" :hide-favorited-and-boosted-by="true" :hide-mention-account="true" me--2 />
+      </div>
+      <!-- Content -->
+      <div space-y-3 my-2>
+        <NuxtLink v-if="derivedStatus.url" :to="derivedStatus.url" external target="_blank">
+          <StatusBody v-if="derivedStatus" :status="derivedStatus" :with-action="false" class="font-light" />
+        </NuxtLink>
+      </div>
+      <!-- Attachments -->
+      <div v-if="card.image" space-y-3 my-2>
+        <StatusMedia
+          v-if="derivedStatus.mediaAttachments?.length > 0"
+          :status="derivedStatus"
+          :is-preview="true"
+        />
```

Comment:
I saw a jump on [this one](https://deploy-preview-2125--elk-zone.netlify.app/universeodon.com/@atomicpoet@calckey.social/110421838157818549), the space isn't reserved until the image is loaded. @ayoayco said we have the width/height of the image: https://github.com/elk-zone/elk/pull/456#issuecomment-1374027756, so it looks like we could avoid this layout shift.

Reply:
I can hard-code the image tag like @ayoayco had initially proposed. The drawback is that we lose the interactivity (e.g. click to open lightbox) that is currently enabled by running it through the `StatusMedia` pipeline.

FWIW, it's actually pretty hard to find posts with links to other posts. It's rarer, still, to find posts with links to other posts that include media attachments. When scrolling through a timeline, the jump is virtually imperceptible b/c the loading nearly always happens before the status makes it into the view. Then, once it's loaded the first time, going from timeline -> detail view is seamless* (*on my computer).

^^ That's my pitch for keeping it as-is, but I'm happy to hard-code the attachment if you prefer.

Reply:
I like the idea of going through the `StatusMedia` pipeline, but you could pass the width and height as optional props to the component to avoid the jump, no?

Reply:
AFAICT we have 2 options: we can either tell Elk to treat the attachment as a full-sized image/video by settting `fullSize === true` *or* we can mock the attachment's meta tags programmatically:

<img width="646" alt="code" src="https://github.com/elk-zone/elk/assets/116459476/59a9a893-92a8-4406-ae3c-4a334bb2df3b">

For some reason, I forgot to set `fullSize` so, I'll do that before resubmitting.

FTR: I initially went down the route of mocking the meta tags, but changed my mind once I realized that I would need to create custom mocking logic for each attachment type:

<img width="616" alt="masto_media_meta" src="https://github.com/elk-zone/elk/assets/116459476/cc996f95-228d-4828-9364-9509eb4a8ec8">


That's *a lot* of code to write *just* to avoid making a single Masto API call to fetch the original status, which would get us the actual attachments in addition to all the other attributes that are currently being mocked to make this work without additional API calls.

Reply:
I suppose another option would be to add a div element with  `flex flex-initial basis-[Xpx]` where X === `card.height` but that requires a level of tinkering with `flex` attributes that I'm not prepared to do ATM lol

The jumping bothers me, too, even if it's not always noticeable. I'd like to get these changes done, and revisit in a week or two once my brain is a little less fried than it currently is -- frontend dev work is not my forté lol